Source/CesiumRuntime/Private/CesiumGeoreference.cpp

```diff
@@ -541,14 +562,13 @@ void ACesiumGeoreference::Tick(float DeltaTime) {
         !cameraLocation.Equals(
             FVector(0.0f, 0.0f, 0.0f),
             this->MaximumWorldOriginDistanceFromCamera)) {
-      // Camera has moved too far from the origin, move the origin.
-      this->GetWorld()->SetNewWorldOrigin(FIntVector(
-          static_cast<int32>(cameraLocation.X) +
-              static_cast<int32>(originLocation.X),
-          static_cast<int32>(cameraLocation.Y) +
-              static_cast<int32>(originLocation.Y),
-          static_cast<int32>(cameraLocation.Z) +
-              static_cast<int32>(originLocation.Z)));
+      // Camera has moved too far from the origin, move the origin,
+      // but make sure that no component exceeds the maximum value
+      // that can be represented as a 32bit signed integer.
+      int64 newX = clampedAdd(cameraLocation.X, originLocation.X);
+      int64 newY = clampedAdd(cameraLocation.Y, originLocation.Y);
+      int64 newZ = clampedAdd(cameraLocation.Z, originLocation.Z);
```

Comment:
```suggestion
      int32 newX = clampedAdd(cameraLocation.X, originLocation.X);
      int32 newY = clampedAdd(cameraLocation.Y, originLocation.Y);
      int32 newZ = clampedAdd(cameraLocation.Z, originLocation.Z);
```

It looks like `clampedAdd` returns an int32. This looks good to me otherwise!